### DISCUSSION TITLE
Dashboard widget layouts for member roles

### DISCUSSION THREAD

**Discussion by shbchk** [ExpressionEngine/ExpressionEngine#3696] [Feature Requests]
### Suggest an idea for ExpressionEngine

Right now Dashboard layouts are saved on a per-member basis. There is no way to enable or disable CP widgets for all members, or a per-role basis.

Please, give us more control over our user dashboards :)

**Comment by intoeetive** [comment]
Technically, dashboard layout can be associated with Role (there is role_id property), so this is more of UX change

Possibly, we could do the following:
replace "save layout" button with the dropdown that would allow setting current layout as default for certain role, similar to this (I don't like how long the text is though) 
![image](https://github.com/ExpressionEngine/ExpressionEngine/assets/752126/e966b5e2-23f9-4e88-8830-09ee4b27eb7c)

The problem is that if the user is changing the layout for themselves, it would have higher precedence, and if then new layout is introduced by site admins - there is no way for user to switch bach to default layout.
So we might need and extra setting to the profiles - whether to use role layout or personal layout.
And then perhaps we might need a permission layer, whether we want to allow users editing their dashboard, or we want them to stick with what has been defined for role.

So it really needs to be thought through before implementing, but I like the idea

**Comment by nep** [comment]
So here's how I'd do it...

1) There's a default dashboard layout that everyone gets if nothing else is set. This everyone-dashboard can be modified. 

2) If you modify the dashboard, you can save it for yourself, for a role, or for everyone (default) (if you have that write-layout-permission-level) -- I like your dropdown, I'd say:
* Save for me
* Save for Super Admin role
* Save for Editor role
* Save for everyone

3) The gear menu is also a "switch layout" menu - similar dropdown:
Use my personal layout
Use Super Admin layout
Use Editor layout
Use default layout
Delete this layout
. You can switch between the created layouts available to you (personal, role(s), everyone) -- so you have a preferred layout. There's no "precedence" -- just what you've chosen. When you save a layout, you switch to using that layout. So if you customize the "everyone" layout and save it for yourself, you're now using that layout preferred.

4) If your preferred layout doesn't exist any more or if you haven't picked one, you get your personal layout. If that doesn't exist, you get the saved layout for your primary role, or if none exists, the saved layout for everyone. So there's no precedence, but there is a series of fall-backs.

4) Store the save date of each layout, and also store the date you selected your layout. If there's a layout available to you that's newer than your selection date, give a constant dashboard alert that says "An updated layout is available." and if they switch to it, then obv that message goes away, and if they then switch back away from that, obv the message stays away. 

5) Of course, if your preferred dashboard is updated, you don't get a message, you just keep using the dashboard and get its updated form. Great for role layouts!

6) If you can change a layout, can you delete that layout? Hmm. Sure. Why not? So that means

7) new needed Member Role Permissions:
* Can select a preferred dashboard layout
* Can create/change/delete personal dashboard layout
* Can create/change/delete this role's dashboard layout
* Can create/change/delete all role dashboard layouts
I don't think you need to be able to create but not delete dashboard layouts, or to be able to create layouts for some other roles but not all, etc. etc. etc.

**Comment by CreativePoweredGeek** [comment]
Honestly,  for most of my use, I'd just like the ability to define what each role gets to see. Much like Channel Layouts... 
I personally don't know that allowing them so save their own dashboard layout is necessary...  I would need to make a lot of different widgets for that to be useful, and honestly, I would just like to control that.
